logpush.go

```diff
@@ -0,0 +1,122 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// LogpushJob describes a Logpush job
+type LogpushJob struct {
+	ID              int        `json:"id,omitempty"`
+	Enabled         bool       `json:"enabled"`
+	Name            string     `json:"name"`
+	LogpullOptions  string     `json:"logpull_options"`
+	DestinationConf string     `json:"destination_conf"`
+	LastComplete    *time.Time `json:"last_complete,omitempty"`
+	LastError       *time.Time `json:"last_error,omitempty"`
+	ErrorMessage    string     `json:"error_message,omitempty"`
+}
+
+// LogpushJobsResponse is the API response, containing an array of Logpush Jobs.
+type LogpushJobsResponse struct {
+	Success  bool         `json:"success"`
+	Errors   []string     `json:"errors"`
+	Messages []string     `json:"messages"`
+	Result   []LogpushJob `json:"result"`
+}
+
+// LogpushJobDetailsResponse is the API response, containing a single Logpush Job.
+type LogpushJobDetailsResponse struct {
+	Success  bool       `json:"success"`
+	Errors   []string   `json:"errors"`
+	Messages []string   `json:"messages"`
+	Result   LogpushJob `json:"result"`
+}
+
+// CreateLogpushJob creates a new LogpushJob for a zone.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-create-logpush-job
+func (api *API) CreateLogpushJob(zoneID string, job LogpushJob) (*LogpushJob, error) {
+	uri := "/zones/" + zoneID + "/logpush/jobs"
+	res, err := api.makeRequest("POST", uri, job)
+	if err != nil {
+		return nil, errors.Wrap(err, errMakeRequestError)
+	}
+	var r LogpushJobDetailsResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, errors.Wrap(err, errUnmarshalError)
+	}
+	return &r.Result, nil
+}
+
+// ListLogpushJobs returns all Logpush Jobs for a zone.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+func (api *API) ListLogpushJobs(zoneID string) ([]LogpushJob, error) {
```

Comment:
The other functions for returning all of the resources that have been added recently omit the `List` bit and just use the plural of the individual fetch function. Here, it would be `LogpushJobs`.